sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -338,159 +348,293 @@ export class StreamingReceiver extends MessageReceiver {
     };
   }
 
+  private _reportInternalError(error: Error): Promise<void> {
+    const messageHandlers = this._messageHandlers();
+
+    if (messageHandlers.forwardInternalErrors) {
+      const errorArgs: InternalProcessErrorArgs = {
+        error,
+        entityPath: this.entityPath,
+        errorSource: "internal",
+        fullyQualifiedNamespace: this._context.config.host
+      };
+
+      return messageHandlers.processError(errorArgs as ProcessErrorArgs);
```

Comment:
This is the user's error handler right? Might want to add a catch and log so an error thrown from the user's code doesn't blow us up.

Reply:
Just above the 'if' block you can see I'm getting these from the "safe" this._messageHandlers(), which means they've been wrapped.

The way I've rewritten this it's now no longer possible to get access to the originally passed in messageHandlers - it's safe messageHandlers or _nothing_!